src/sql/connection.py

```diff
@@ -552,6 +552,10 @@ def _get_curr_sqlglot_dialect(self):
             Available dialect in sqlglot package, see more:
             https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/dialect.py
         """
+
```

Comment:
the name is a property that identifies the connection, it's unrelated to the database type

Reply:
- Can I use `self.dialect` of class `DBAPIConnection`?
Also, `self.dialect` and `self.name` both are the same and equal to `connection_name_`, so I don't quite get the use of these variables.

- I can also call `_check_if_duckdb_dbapi_connection` again in `_get_curr_sqlglot_dialect`

The bottom line is that I need `_get_curr_sqlglot_dialect` to return `"duckdb"` instead of `None`, so that the query can be transpiled appropriately.

Reply:
All the functions and variables are in [connections.py](https://github.com/ploomber/jupysql/blob/master/src/sql/connection.py)

Reply:
@tonykploomber @edublancas 
Please correct me if I am wrong, but In the case of the custom connectors using `DBAPIConnection` will not work properly as the `_get_curr_sqlglot_dialect` function will return `None` and `sqlglot` will transpile it accordingly. For example: 

```SQL

    SELECT
    percentile_disc([0.25, 0.50, 0.75]) WITHIN GROUP     (ORDER BY "body_mass_g") AS percentiles,
    AVG("body_mass_g") AS mean,
    COUNT(*) AS N
    FROM "penguins"
```

will transpiled to 

```SQL
SELECT PERCENTILE_DISC(ARRAY(0.25, 0.50, 0.75)) WITHIN GROUP (ORDER BY "body_mass_g") AS percentiles, AVG("body_mass_g") AS mean, COUNT(*) AS N FROM "penguins"
```

whether it is valid for custom connectors or not(This effect is irrespective of this PR and issue) 

### Possible Solution
- Completely remove the transpile query for `DBAPIConnection`, making it dependent on the user to pass the correct query for the connector 
OR
- Add support for more native Python connectors and use transpile and remove transpile for custom ones.

PS: It seems like there is a separate issue #732  and a PR #747 which is handling `_get_curr_sqlglot_dialect`. So I am not quite sure, what will be the scope of my PR, as solving #732 will solve boxplot issue and histogram is working irrespective of this PR

Reply:
> Can I use self.dialect of class DBAPIConnection?

yes you can use it

> Also, self.dialect and self.name both are the same and equal to connection_name_, so I don't understand the use of these variables

we have to clean up this code, as you can see, there is some redundancy